src/GitVersion.Core/Core/GitPreparer.cs

```diff
@@ -68,7 +68,13 @@ private void PrepareInternal(bool normalizeGitDirectory, string? currentBranch,
             }
             else
             {
-                if (!normalizeGitDirectory) return;
+                if (!normalizeGitDirectory)
+                {
+                    if (currentBranch == null) return;
+                    EnsureLocalBranchExistsForCurrentBranch(EnsureOnlyOneRemoteIsDefined(), currentBranch);
```

Comment:
It seems weird to me to perform normalization even to `/nonormalize` is being passed as an argument. Isn't it better to ensure that the Git repository is normalized before GitVersion is executed?

Reply:
GitVersion has low performance with many branches. I found out, that the normalization was the reason why it takes minutes.
But with the /nonormalize parameter GitVersion did not work when it is used on a buildserver.

Reply:
`/nonormalize` works on build servers if you normalize the repository yourself. [GitVersion's operation requirements are found in the documentation](https://gitversion.net/docs/reference/requirements). When the repository is normalized to meet those requirements, it should execute successfully in any environment.